packages/scene-composer/src/components/panels/scene-settings/ComponentVisibilityToggle.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useContext, useMemo } from 'react';
+import { useIntl } from 'react-intl';
+import { Box, Toggle } from '@awsui/components-react';
+
+import { IDataOverlayComponentInternal, useStore, useViewOptionState } from '../../../store';
+import { sceneComposerIdContext } from '../../../common/sceneComposerIdContext';
+import { KnownComponentType } from '../../../interfaces';
+import { Component } from '../../../models/SceneModels';
+import { findComponentByType } from '../../../utils/nodeUtils';
+
+export interface ComponentVisibilityToggleProps {
+  componentType: KnownComponentType | Component.DataOverlaySubType;
+  label: string;
+  onChange?: (newValue: boolean) => void;
+}
+
+export const ComponentVisibilityToggle: React.FC<ComponentVisibilityToggleProps> = ({
```

Comment:
mostly the toggle will only be on the scene's settings page since it can only handle visibility of all instances of the components of the specified type, but not for one instance, so I'll just place it here. we can move it later if other panels want it